### ISSUE TITLE
Extended Unicode key  Menu when Holding down a key

### ISSUE THREAD

**Issue by RamonUnch** [RamonUnch/AltSnap#181]
I got the request from some people to add the ability to mimic a MacOS option, also present in android, where when you hold down a key, instead of auto-repeating it, you get a list of extra characters that you can type instead of the initial character. I implemented this option in AltSnap, it can be enabled using this (beta) version below in the **Keyboard tab** you can check the **Popup an extended character list menu when holding an alphabetic key down** option.
Exemple:
![unikeymenu](https://user-images.githubusercontent.com/74856804/159156977-244cb9f3-eb45-4c20-b1f8-115c282a2d4b.png)

To enable the desired extended key I suggest you hit the accelerator key (A-Z) . You can also select the menu item with arrow keys and with the mouse. Press ESC/Enter/Space/Backspace or the hotkey if you want to hide the menu.

**NOTES:**
1) Extra key list can be setup as a per-language option, you just need to add the list in the \lang_COUNTRY.ini` file, more details in the next post.
2) It is not always possible to find the caret position in a window, for now I fallback to the mouse cursor position.
Browsers window, for example, will not behave properly because they do not use a native caret.
3) Only the virtual keys from A to Z are supported for extra chars, this may change.
4) If you make a capital (with caps lock or shift) you will have a popup menu with the uppercase equivalents (if applicable only ie: é->É, ǳ->Ǳ but ǲ has no uppr/lower case transformation, for the dotted capital I **İ** and the dot-less small i **ı** there are also no upper/lower case conversions). It is a bit of a mess because the `CharUpperW()` function cannot handle a lot of Latin Extended-B (0180 - 024F) characters, such as **Ș**, those one have to be manually handled...
5) when you press space/backspace/the hotkey or escape the menu will pop down.

In the long run, this may become a separated application. For now I think that adding it to AltSanp just adds 2KB to the exe and does not hurt anyone. There is a per-language config so that non-western people can take full advantage if the option. However the list is only populated with extra Latin characters.

EDIT: new test build with some fixes and per-language config plus support for extended Unicode plane.
**You need to erase `AltSnap.exe` and `hooks.dll` in your old install directory:**
[AltSnap1.53test14_i386.zip](https://github.com/RamonUnch/AltSnap/files/8360511/AltSnap1.53test14_i386.zip)
[AltSnap1.53test14_x64.zip](https://github.com/RamonUnch/AltSnap/files/8360514/AltSnap1.53test14_x64.zip)

**Comment by RamonUnch** [comment]
New verbs in the translation file:
You can modify all or a selection of the A-Z letters for a given language. Probably it would be better to have a list depending on the current keyboard layout but I will see this for later. You can use any Unicode characters even in the extended plane
Encoding has to be UTF16-LE (two surrogates for extended plane), so you can include emoji, even though they are already pretty easy to do on Window 10 (Window+; shortcut).
A good starting point to see and copy/paste extra characters is: https://unicode-table.com/

```
InputUniKeyHoldMenu=Pop&up an extended character menu when holding an alphabetic key down

; List of characters that populate the extended key menu
A=àáâäãåæª%āăąǎǟǡǣǻǽȁȃȧ|Ȧḁ%ⱥ|Ⱥɐ|Ɐɑ|Ɑɒ|Ɒⲁ|Ⲁ
B=%ƀɓƃƅɃ%ɓ|Ɓḃḅḇ
C=ç¢©%ćĉċčƈḉȼ|Ȼɕ
D=ð%ďđɖɗƌƍḋḍḏḑḓǆǅǳǲȡȸ
E=èéêë€%ēĕėęěǝəɛȅȇḕḗḙḛȩ|Ȩḝɇ|Ɇⱸ
F=ƒ%ḟɸⱷ
G=%ǵǧḡɠɣǥ
H=%ĥħƕǶḣḥḧḩḫȟ|Ȟⱨ|Ⱨⱶ|Ⱶẖ
I=ìíîï%ĩīĭǐȉȋįİıĳɩɨḭḯ
J=%ĵǰȷɉ|Ɉ
K=%ķĸƙǩḱḳⱪ|Ⱪꝁ|Ꝁʞ|Ʞ
L=£%ĺļľŀłƛǉǈȴƚ|Ƚⱡ|Ⱡɫ|Ɫḷḹḻḽ
M=µ%ḿṁṃɱ|Ɱɯ
N=ñ%ńņňŉŋɲƞ|Ƞǌǋǹȵ%ṅṇṉṋ
O=òóôöõø°%ōŏő%ɔɵơƣǒǫǭǿȍȏȣ|Ȣȫ|Ȫȭ|Ȭȯ|Ȯȱ|Ȱṍṏṑṓ%ⱺ
P=¶þ·•%ƥᵽ|Ᵽṕṗ
Q=¿¤‰‘’“”„…–—«»‹›%ȹɋ|Ɋ
R=®%ŕŗřƦȑȓṙṛṝṟɍ|Ɍɽ|Ɽⱹ
S=šß§%śŝşſ%ƨʃƪș|Șȿ|Ȿ%ṡṣṥṧṩ
T=†‡™%ţťŧƫƭʈț|Țȶⱦ|Ⱦ%ṫṭṯṱẗ
U=ùúûü%ũūůŭűų%ưʊǔǖǘǚǜȕȗʉ|Ʉ%ṳṵṷṹṻ
V=%ʋɅⱱⱴ%ṽṿ
W=%ẁẃŵẅⱳ|Ⱳ%ẇẉ
X=±×÷¬%ẋẍ
Y=ýÿ¥%ŷẏȳ|Ȳƴɏ|Ɏ
Z=ž%źẑżẓẕ%ƶʒƹƺǯȥ|Ȥɀ|Ɀⱬ|Ⱬ
```
* No separator between each character, just the list...
* You can put a % to include a separator, I used it to separate LATIN-1 (cp1252) from Unicode characters.
* All char should be set lower-case if there is a lower->upper conversion possible (see below).
* Use `lowercase|UPPERCASE` if windows can not convert the letter by himself ie: `a|Ab|Bc|C`. In general it is not needed but Windows is buggy for some characters. For instance It is required for `ȧ|Ȧ`, `ȭ|Ȭ` but not needed for `ṓ` ask MS why...
it is not a problem if you over-do it and specify `x|X` even if x's upper-casing is handled by `CharUpper()`, but you will do less mistakes if you minimize the effort.

**Comment by RamonUnch** [comment]
Feature is available in 1.53
Open new issues for future problems.
